horovod/mxnet/mpi_ops.cc

```diff
@@ -93,35 +74,16 @@ void DoAllgather(NDArray* tensor, NDArray* output, std::string& name,
   auto hvd_tensor = std::make_shared<MXTensor<NDArray>>(tensor);
   auto hvd_context = std::make_shared<MXOpContext<NDArray>>(device, output);
 
-  auto enqueue_result =
-      EnqueueTensorAllgather(hvd_context, hvd_tensor, nullptr,
-                             name, device,
-                             [on_complete](const Status& status) {
-                               InvokeCompleteCallback(on_complete, status);
-                             });
-  ThrowIfError(enqueue_result);
-}
-
-#if HAVE_CUDA
-void DoAllgatherCudaOnCPU(MXTempBufferShared& hvd_cpu_buffer, std::string& name,
-                          CallbackOnComplete on_complete) {
-  ThrowIfError(common::CheckInitialized());
-
-  auto hvd_context = std::make_shared<MXOpContext<NDArray>>(
-      CPU_DEVICE_ID, hvd_cpu_buffer->tensor());
-
   auto enqueue_result = EnqueueTensorAllgather(
-      hvd_context, hvd_cpu_buffer, nullptr,
-      name, CPU_DEVICE_ID,
+      hvd_context, hvd_tensor, nullptr, name, device,
       [on_complete](const Status& status) {
         InvokeCompleteCallback(on_complete, status);
       });
   ThrowIfError(enqueue_result);
 }
-#endif
 
-void DoBroadcast(NDArray* tensor, NDArray* output, int root_rank,
-                 std::string& name, CallbackOnComplete on_complete) {
+void DoBroadcast(NDArray* tensor, NDArray* output, std::string& name,
```

Comment:
const std::string&name?

Reply:
Let's not worry about this. This function will be changed in the other PR solving the GCC compatibility issue.